lib/src/collection/kt_list.dart

```diff
@@ -183,6 +183,28 @@ extension KtListExtensions<T> on KtList<T> {
     }
   }
 
+  /// Returns the first non-null value after applying the given [transform]
+  /// function, throwing a [NoSuchElementException] exception if there is no
+  /// such value.
+  R firstNotNullOf<R>(R? Function(T?) transform) {
+    final KtList<R> mappedList = mapNotNull(transform);
+
+    if (mappedList.isEmpty()) {
+      throw const NoSuchElementException(
+        "Collection contains no element different than null after transform.",
+      );
+    }
+
+    return mappedList.first();
```

Comment:
nit: Any reason you differ from the [kotlin impl](https://github.com/JetBrains/kotlin/blob/7060811c8b25678e7046f9fb569a8f0f1f782100/libraries/stdlib/common/src/generated/_Collections.kt#L236) by not calling `firstNotNullOfOrNull`? Calling `isEmpty` and then `first` requires to check for an existing value twice, whereas `firstNotNullOfOrNull` only checks once

Reply:
that's a good point. I'm gonna implement this way